src/extension.ts

```diff
@@ -36,7 +66,22 @@ export function activate(context: vscode.ExtensionContext) {
     const { isActive } = await repository.get();
 
     if (isActive) {
+      if (useMdPreviewSetting && curSlideLanguageId === "markdown") {
+        await vscode.commands.executeCommand(
+          "workbench.action.closeActiveEditor"
+        );
+        await vscode.workspace.openTextDocument(curSlideUri);
+      }
+
       await vscode.commands.executeCommand("workbench.action.nextEditor");
+      let activeWindow = vscode.window.activeTextEditor;
+
+      if (useMdPreviewSetting && activeWindow !== undefined) {
+        curSlideLanguageId = activeWindow.document.languageId;
+        if (activeWindow.document.languageId === "markdown") {
+          await vscode.commands.executeCommand("markdown.showPreview");
+        }
+      }
```

Comment:
👍 Thank you, that works well!

I think I'll refactor the code. I'd love to extract the domain logic from the VS Code API, so I can test that and remove some code duplication too. I'd do that in follow-up commits, don't worry about it.

I also noticed there is a bit of a glitch when you move from a Markdown to another. It will close the preview, open the markdown, then show the new preview. We temporarily see the Markdown file, very briefly. I think we could optimise this. I think I can also do this after I refactor the code.